parsing/parsetree.mli

```diff
@@ -165,541 +182,584 @@ and row_field = {
-  | Pctf_inherit of class_type
-        (* inherit CT *)
+  | Pctf_inherit of class_type  (** [inherit CT] *)
   | Pctf_val of (label loc * mutable_flag * virtual_flag * core_type)
-        (* val x: T *)
-  | Pctf_method  of (label loc * private_flag * virtual_flag * core_type)
-        (* method x: T
+      (** [val x: T] *)
+  | Pctf_method of (label loc * private_flag * virtual_flag * core_type)
+      (** [method x: T]
 
-           Note: T can be a Ptyp_poly.
-         *)
-  | Pctf_constraint  of (core_type * core_type)
-        (* constraint T1 = T2 *)
-  | Pctf_attribute of attribute
-        (* [@@@id] *)
-  | Pctf_extension of extension
-        (* [%%id] *)
-
-and 'a class_infos =
-    {
-     pci_virt: virtual_flag;
-     pci_params: (core_type * (variance * injectivity)) list;
-     pci_name: string loc;
-     pci_expr: 'a;
-     pci_loc: Location.t;
-     pci_attributes: attributes;  (* ... [@@id1] [@@id2] *)
-    }
-(* class c = ...
-   class ['a1,...,'an] c = ...
-   class virtual c = ...
-
-   Also used for "class type" declaration.
+            Note: [T] can be a {{!core_type_desc.Ptyp_poly}[Ptyp_poly]}.
+        *)
+  | Pctf_constraint of (core_type * core_type)  (** [constraint T1 = T2] *)
+  | Pctf_attribute of attribute  (** [[\@\@\@id]] *)
+  | Pctf_extension of extension  (** [[%%id]] *)
+
+and 'a class_infos = {
+  pci_virt : virtual_flag;
+  pci_params : (core_type * (variance * injectivity)) list;
+  pci_name : string loc;
+  pci_expr : 'a;
+  pci_loc : Location.t;
+  pci_attributes : attributes;  (** [... [\@\@id1] [\@\@id2]] *)
+}
+(** Values of type [class_expr class_infos] represents:
+    - [class c = ...]
+    - [class ['a1,...,'an] c = ...]
+    - [class virtual c = ...]
+
+   They are also used for "class type" declaration.
 *)
 
 and class_description = class_type class_infos
-
 and class_type_declaration = class_type class_infos
 
-(* Value expressions for the class language *)
+(** {2 Value expressions for the class language} *)
 
-and class_expr =
-    {
-     pcl_desc: class_expr_desc;
-     pcl_loc: Location.t;
-     pcl_attributes: attributes; (* ... [@id1] [@id2] *)
-    }
+and class_expr = {
+  pcl_desc : class_expr_desc;
+  pcl_loc : Location.t;
+  pcl_attributes : attributes;  (** [... [\@id1] [\@id2]] *)
+}
 
 and class_expr_desc =
   | Pcl_constr of Longident.t loc * core_type list
-        (* c
-           ['a1, ..., 'an] c *)
-  | Pcl_structure of class_structure
-        (* object ... end *)
+      (** [c] and [['a1, ..., 'an] c] *)
+  | Pcl_structure of class_structure  (** [object ... end] *)
   | Pcl_fun of arg_label * expression option * pattern * class_expr
-        (* fun P -> CE                          (Simple, None)
-           fun ~l:P -> CE                       (Labelled l, None)
-           fun ?l:P -> CE                       (Optional l, None)
-           fun ?l:(P = E0) -> CE                (Optional l, Some E0)
-         *)
+      (** [Pcl_fun(lbl, exp, P, CE)] represents:
```

Comment:
`exp` is expanded to `Some E0`, it will suggest to use `exp0` to keep indices consistent.